cassandra-tests/src/main/java/cassandra/ForwardHttpSpansToHandler.java

```diff
@@ -0,0 +1,158 @@
+ */
+// Temporary until moved into brave-tests.
+public class ForwardHttpSpansToHandler implements TestRule {
+  final MockWebServer zipkin = new MockWebServer();
+
+  /**
+   * The port variable in {@code POST http://localhost:${port}/api/v2/spans}.
+   *
+   * <p><h3>Testcontainers note</h3>
+   * This is typically used to allow instrumentation in Docker containers to be tested the same as
+   * in- memory. There's a catch to using this in Testcontainers 1.15, which is you have to clear
+   * the port prior to starting the container.
+   *
+   * <pre>${code
+   *  // Allows http://host.testcontainers.internal:" + zipkin.httpPort() + "/api/v2/spans"
+   *  Testcontainers.exposeHostPorts(zipkin.httpPort());
+   * }</pre>
+   */
+  public int httpPort() {
+    return zipkin.getPort();
+  }
+
+  public ForwardHttpSpansToHandler(SpanHandler spanHandler) {
+    if (spanHandler == null) throw new NullPointerException("spanHandler == null");
+    zipkin.setDispatcher(new ZipkinDispatcher(spanHandler));
+  }
+
+  @Override public Statement apply(Statement base, Description description) {
+    return zipkin.apply(base, description);
+  }
+
+  static final class ZipkinDispatcher extends Dispatcher {
+    final SpanHandler spanHandler;
+
+    ZipkinDispatcher(SpanHandler spanHandler) {
+      this.spanHandler = spanHandler;
+    }
+
+    @Override public MockResponse dispatch(RecordedRequest request) {
+      if ("POST".equals(request.getMethod())) {
+        String type = request.getHeader("Content-Type");
+        if ("/api/v2/spans".equals(request.getPath())) {
+          SpanBytesDecoder decoder = type != null && type.contains("/x-protobuf")
+              ? SpanBytesDecoder.PROTO3
+              : SpanBytesDecoder.JSON_V2;
+          return acceptSpans(request, decoder);
+        }
+      } else { // unsupported method
+        return new MockResponse().setResponseCode(405);
+      }
+      return new MockResponse().setResponseCode(404);
+    }
+
+    MockResponse acceptSpans(RecordedRequest request, SpanBytesDecoder decoder) {
+      byte[] body = request.getBody().readByteArray();
+      String encoding = request.getHeader("Content-Encoding");
+      if (encoding != null && encoding.contains("gzip")) {
+        try {
+          Buffer result = new Buffer();
+          GzipSource source = new GzipSource(new Buffer().write(body));
+          while (source.read(result, Integer.MAX_VALUE) != -1) ;
+          body = result.readByteArray();
+        } catch (IOException e) {
+          return new MockResponse().setResponseCode(400).setBody("Cannot gunzip spans");
+        }
+      }
+
+      if (body.length == 0) return new MockResponse().setResponseCode(202); // lenient on empty
+
+      for (Span span : decoder.decodeList(body)) {
+        spanHandler.end(null, toMutableSpan(span), SpanHandler.Cause.FINISHED);
+      }
+
+      return new MockResponse();
+    }
+  }
+
+  static MutableSpan toMutableSpan(Span span) {
```

Comment:
@jcchavezs I think code like this is why people decide to pin to completely new collector architectures. converting data types is.. sooo hard :P